runtime/src/main/java/com/dylibso/chicory/runtime/StackFrame.java

```diff
@@ -104,9 +104,9 @@ public void dropValuesOutOfBlock(MStack stack) {
             }
 
             int expectedStackSize = stackSizeBeforeBlock.pop();
-            while (stack.size() > expectedStackSize) {
-                stack.pop();
-            }
+            // todo: the expected stack size is often larger than the actual stack; investigate &
```

Comment:
This todo is a bit worrying and the current implementation is probably just hiding a bug somewhere else. I'd probably investigate this further before merging this.

Reply:
This patch didn't cause this issue, but it did uncover it. I changed the old pop loop, which pops until the expected size, to pop to the previous depth exactly (see the comment about removing `Math.min`), and it cause failures. So I added in the `Math.min` to replicate the previous behavior for later investigation.

Reply:
I guess the original problem is subtle. The intent of the loop above is "while the stack is bigger than the expected size, pop", but it also passes quietly if the stack size is *smaller* than the expected size. This subtle behavior is what went undetected, and it's almost certainly incorrect.